armi/operators/operator.py

```diff
@@ -618,6 +633,51 @@ def interactAllCoupled(self, coupledIteration):
         activeInterfaces = [ii for ii in self.interfaces if ii.enabled()]
         self._interactAll("Coupled", activeInterfaces, coupledIteration)
 
+    def _tightCouplingOldValues(self):
+        activeInterfaces = [ii for ii in self.interfaces if ii.enabled()]
+        for interface in activeInterfaces:
+            interface.tightCouplingOldValue = interface.getTightCouplingValue()
+
+    def _computeTightCouplingConvergence(self):
+        activeInterfaces = [ii for ii in self.interfaces if ii.enabled()]
+        convergence = []
+        for interface in activeInterfaces:
+            if interface.tightCouplingOldValue is not None:
+                key = interface.name+": "+interface.tightCouplingConvergeOn
+                if isinstance(interface.tightCouplingOldValue, float):
+                    eps = abs(interface.tightCouplingOldValue - interface.getTightCouplingValue())
+                elif isinstance(interface.tightCouplingOldValue, ndarray):
+                    newValue = interface.getTightCouplingValue()
+                    epsVec = []
+                    for old,new in zip(interface.tightCouplingOldValue, newValue):
+                        epsVec.append(norm(subtract(old,new), ord=2))
+                    eps = norm(epsVec, ord=inf)
+                else:
+                    raise RuntimeError("Only currently set up to handle either floats or lists... Sorry.")
+                self.convergenceSummary[key].append(eps)
+                if eps > interface.tightCouplingTolerance:
+                    convergence.append(False)
+                else:
+                    convergence.append(True)
```

Comment:
Something tells me that this can be cleaned up a bit using local variables. Additionally, the `key` local variable might be better put together with a `.join` call.

Reply:
Here's part 1 with `.join`. Working on cleaning this up.
https://github.com/terrapower/armi/pull/1033/commits/f8f8a8e9a1d71c5df61dad321d40fd9f3cd01db8